### DISCUSSION TITLE
Handle two modals with two event buses

### DISCUSSION THREAD

**Discussion by davidgarra** [mia-platform/community#377] [Q&A] [answered]
### Which section?

Backoffice

### Which is the version?

back-kit@1.4.7, microlc-backend@0.8.1, microlc-frontend@0.9.0

### Description

Hi everyone,
I'd like to have 2 different modals on the same page. To achieve this, I'm using the event-bus. I've created the following components:
1. `bk-button` having a specific event-bus "eventBus.pool.sts"
```json
{
  "type": "element",
  "tag": "bk-button",
  "properties": {
    "eventBus": "eventBus.pool.sts",
    "content": "test",
    "action": {
      "type": "event",
      "config": {
        "events": {
          "label": "selected-data",
          "payload": {}
        }
      }
    }
  }
}
```
2. `bk-add-new-button` with no event-bus defined
```json
{
  "type": "element",
  "tag": "bk-add-new-button"
}
```
3. `bk-dynamic-form-modal` having a specific event-bus "eventBus.pool.sts"
```json
{
  "type": "element",
  "tag": "bk-dynamic-form-modal",
  "properties": {
    "eventBus": "eventBus.pool.sts",
    "dataSchema": {
      "type": "object",
      "properties": {
        "_id_": {
          "type": "string",
          "formOptions": {
            "hidden": true
          }
        },
        "name": {
          "type": "string"
        },
        "price": {
          "type": "number"
        }
      }
    }
  }
}
```
4. `bk-form-modal` with no event-bus defined
```json
{
  "type": "element",
  "tag": "bk-form-modal",
  "properties": {
    "wizard": [
      ... // here there are all the wizard props
    ],
    "dataSchema": {
      "$ref": "dataSchema"
    },
    "requireConfirm": true,
    "allowObjectAsTable": true
  }
}
```

The current problem is that both buttons open both modals...
If I click on "Add new," both modals open, one above the other.
If I click on "Test," both modals open, one above the other.

### Actual Outcome

Any button opens both modals

### Expected Outcome

Each button should open only one modals

**Comment by clabene** [comment, marked as answer]
Hi @davidgarra 

since you are using micro-lc version < 2.0, property `eventBus` is not available. Instead, you should use `busDiscriminator` key at the same level of `tag` and `properties`.

You should update your configuration to:
```
{
  "type": "element",
  "tag": "bk-button",
  "busDiscriminator": "event-bus-name",
  "properties": {
    ...
  }
}
```

https://micro-lc.io/1.x/docs/core_plugins/#busdiscriminator

**Reply by davidgarra** [reply]
It works like a charm, thank you so much!!!

**Reply by clabene** [reply]
Thanks for the feedback!
